src/views/ChooseObject.vue

```diff
@@ -121,19 +122,37 @@ export default class ChooseObject extends Vue {
     let name = objects[index].name;
     const data = { selection: name };
 
-    const ok = await Swal.fire({
-      title: "Your selection",
-      text: "You have selected " + name,
-      reverseButtons: true,
-    });
-
-    if (ok) {
-      await post<any>("/api/select", data);
-      this.yourSelection = ObjectClass.getImage(objects[index]);
-      this.characterSelected = true;
-      this.displayCommand = false;
-      this.guess();
+    let userGuess = "";
+
+    if(!this.characterSelected) {
+      const ok = await Swal.fire({
+        title: "Your selection",
+        text: "You have selected " + name,
+        reverseButtons: true,
+      });
+
+      if (ok) {
+        await post<any>("/api/select", data);
+        this.yourSelection = ObjectClass.getImage(objects[index]);
+        this.characterSelected = true;
+        this.displayCommand = false;
+        this.guess();
+      }
+    } else {
```

Comment:
This works, but whenever you come accross if/else statements, especially if there is an if inside an if, you should ask yourself if one can extract the code into separate methods. This is for better readability. In this case you might have a method `displaySelection` for the first if, and a method `confirmSelection` for the else statement. Can you please try extracting the code?

Reply:
Thank you for the suggestion. I have worked on it, please let me know what you think.